staking/transaction/transaction.go

```diff
@@ -0,0 +1,111 @@
+package transaction
+
+import (
+	"errors"
+	"io"
+	"math/big"
+	"sync/atomic"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/harmony-one/harmony/crypto/hash"
+	"github.com/harmony-one/harmony/staking/message"
+)
+
+type txdata struct {
+	message.Directive
+	Stake        interface{}
+	AccountNonce uint64   `json:"nonce"      gencodec:"required"`
+	Price        *big.Int `json:"gasPrice"   gencodec:"required"`
+	GasLimit     uint64   `json:"gas"        gencodec:"required"`
+	// Signature values
+	V *big.Int `json:"v" gencodec:"required"`
+	R *big.Int `json:"r" gencodec:"required"`
+	S *big.Int `json:"s" gencodec:"required"`
+	// This is only used when marshaling to JSON.
+	Hash *common.Hash `json:"hash" rlp:"-"`
+}
+
+// Stake is a record captuing all staking operations
+type Stake struct {
+	data txdata
+	// caches
+	hash atomic.Value
+	size atomic.Value
+	from atomic.Value
+}
+
+type fulfill func() (message.Directive, interface{})
```

Comment:
This thing is confusing too. Better to just make Directive and payload as two params in NewStake()

Reply:
but as API consumer, (as hmy CLI is/will be), nicer to have one call to NewStake with different callbacks that I can dynamically make. That is, it decouples things for end user.